### PR TITLE
Add camera lens field of view settings [3d] 

### PR DESCRIPTION
**Pull request by nirvn**
## Description
Being able to set the camera lens' field of view is quite useful in a 3D environment. Qt3D has it, and it's just a no brainer implementation. So here it is :)

This is a QGIS 3D scene, with the default field of view of 45 degree angle:

Here's the same scene, rendered using a field of view of 85 degree angle:

The perspective change can be subtle, but really handy.

@wonder-sk , review required :)

Also, we realllly need to do something about the settings window.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
Nice feature!

Question: did you consider exposing this as a focal length setting?

**Comment by nirvn** [comment]
@nyalldawson , I'm not sure how easy that'd be in this context. I considered it but declined to investigate further after a brief look into it.

**Comment by nyalldawson** [comment]
@nirvn what's blender use for this?

**Comment by nirvn** [comment]
Blender offers both choices.

File: src/3d/qgs3dmapscene.cpp [hunk 1]
```diff
@@ -486,6 +488,16 @@ void Qgs3DMapScene::updateLights()
   }
 }
 
+void Qgs3DMapScene::updateCameraLens()
+{
+  QRect viewportRect( QPoint( 0, 0 ), mEngine->size() );
+  float aspectRatio = ( float )viewportRect.width() / viewportRect.height();
+  mEngine->camera()->lens()->setFieldOfView( mMap.fieldOfView() );
+  mEngine->camera()->lens()->setPerspectiveProjection( mMap.fieldOfView(), aspectRatio, 10.f, 10000.0f );
```
Comment (wonder-sk) [inline src/3d/qgs3dmapscene.cpp]: why not just `mEngine->camera()->lens()->setFieldOfView( ... )` ? We should not need to reset near/far planes and recalculate aspect ratio.  The setPerspectiveProjection() call should be just a shortcut to set multiple parameters at the same time...

File: src/3d/qgs3dmapscene.cpp [hunk 2]
```diff
@@ -71,7 +71,8 @@ Qgs3DMapScene::Qgs3DMapScene( const Qgs3DMapSettings &map, QgsAbstract3DEngine *
 
   // Camera
   float aspectRatio = ( float )viewportRect.width() / viewportRect.height();
-  mEngine->camera()->lens()->setPerspectiveProjection( 45.0f, aspectRatio, 10.f, 10000.0f );
+  mEngine->camera()->lens()->setFieldOfView( mMap.fieldOfView() );
+  mEngine->camera()->lens()->setPerspectiveProjection( mMap.fieldOfView(), aspectRatio, 10.f, 10000.0f );
```
Comment (wonder-sk) [inline src/3d/qgs3dmapscene.cpp]: here it should not be necessary to call setFieldOfView() when it is set again on the following line...

File: src/3d/qgslayoutitem3dmap.cpp [hunk 3]
```diff
@@ -23,6 +23,7 @@
 #include "qgslayoutitemregistry.h"
 #include "qgsoffscreen3dengine.h"
 
+#include <Qt3DRender/QCamera>
```
Comment (wonder-sk) [inline src/3d/qgslayoutitem3dmap.cpp]: probably this include is not needed if there's no other code change in this file?

**Review by wonder-sk — APPROVED** [review]
Nice addition!

**Comment by nirvn** [comment]
@wonder-sk , review addressed, test case added.

**Comment by NathanW2** [comment]
Holy heck! NICE!

**Comment by wonder-sk** [comment]
@nirvn thanks... Ready to merge!

**Pull request merged**